src/classes/BGE_Wizard_CTRL.cls

```diff
@@ -0,0 +1,307 @@
+                'HomeAddressImportStatus__c',
+                'HouseholdAccountImported__c',
+                'ImportedDate__c',
+                'NPSP_Data_Import_Batch__c',
+                'PaymentImported__c',
+                'PaymentImportStatus__c',
+                'Payment_Possible_Matches__c',
+                'Status__c',
+                'Account2ImportStatus__c'
+        };
+
+        Set<String> allowedObjects = new Set<String>{
+                'Opportunity',
+                'Payment'
+        };
+
+        // these fields are required only if at least 1 field from its object group is selected
+        Set<String> systemRequiredFields = new Set<String>{
+                'Account1_Name__c',
+                'Account2_Name__c',
+                'Donation_Amount__c',
+                'Contact1_LastName__c',
+                'Contact2_LastName__c'
+        };
+
+        // looping through the fields on DataImport__c
+        for (Schema.DescribeFieldResult dfr : fieldMap.values()) {
+            String dataImportFieldApiName = UTIL_Namespace.StrTokenRemoveNSPrefix(dfr.name);
+            //Special case for Donation_Donor because it's derived
+
+            //todo: rework these 2 lines
+            String dataImportFieldMappedObject = dfr.inlineHelpText != null ? dfr.inlineHelpText.split('\\.')[0] : null;
+            Boolean fieldAllowed = allowedObjects.contains(dataImportFieldMappedObject) && !bannedFields.contains(dataImportFieldApiName) && dfr.inlineHelpText != null;
+
+            if (fieldAllowed) {
+
+                AvailableField fld = new AvailableField();
+
+                // TODO: make these localized with schema describes
+                fld.sObjectName = dataImportFieldMappedObject;
+
+                // TODO: pull these labels from target fields' schema based on help text mapping
+                fld.label = (dfr.getLabel().contains(' ') && dfr.getLabel().split(' ',2)[0] == fld.sObjectName) ?
+                        dfr.getLabel().split(' ',2)[1] :
+                        dfr.getLabel() ;
+                fld.name = dfr.getName();
+
+                String unnamespacedField = UTIL_Namespace.StrTokenRemoveNSPrefix(fld.name);
+                if (systemRequiredFields.contains(unnamespacedField)) {
+                    fld.systemRequired = true;
+                }
+
+                fields.add(fld);
+            }
+        }
+        return fields;
+    }
+    /*******************************************************************************************************
+    * @description Retrieves the object and field labels for the component.
+    * @param sObjectName the object being used by the component (already stripped of the namespace prefix)
+    * @return Map of object and field name keys to their labels.
+    */
+    public static Map<String, String> getLabels(String sObjectName) {
+
+        if (sObjectName == 'Batch_Template__c') {
+            Map<String, String> labels = new Map<String, String> {
```

Comment:
Split contents of if statements into two separate methods: getBatchTemplateLabels(), getBatchLabels()